docs/genai/data-mgmt/vector-databases.md

```diff
@@ -50,6 +50,228 @@ q_context = " ".join([f"#{str(i)}" for i in results["documents"][0]])
+    for hit in search_results[0]:
+        results.append({"book_id":hit.id, "name":hit.entity.get('book_summaries'), "score":hit.distance})
+        print(f"Book ID: {hit.id}, Name: {hit.entity.get('book_summaries')}, Score: {hit.distance}")
+        
+    return context.Response(body=str(results),
+                            headers={},
+                            content_type='text/plain',
+                            status_code=200)
+```
+
+Build the custom image with the required dependencies and set the functions as part of the project.
+```python
+import mlrun
+project = mlrun.get_or_create_project("book-similarity-milvus", "./")
+
+# Create the image once and use it in every function
+project.build_image(image=".book-similarity-image", base_image="mlrun/mlrun", requirements=["sentence_transformers", "pymilvus"])
+
+# Set the workflow functions
+create_data_func = project.set_function(func="workflow_functions.py", name="create-data", kind="job", image=".book-similarity-image",
+                                        handler="create_data")
+insert_data_into_milvus_func = project.set_function(func="workflow_functions.py", name="insert-data-into-milvus", kind="job", image=".book-similarity-image",
+                                       handler="insert_data_into_milvus")
+
+# Set the inference function (nuclio)
+inference_func = project.set_function(func="inference_function.py", name="inference-function", kind="nuclio", image=".book-similarity-image", 
+                                      handler="inference_function")
+```
+
+Create the workflow file (workflow.py).
+```python
+from kfp import dsl
+
+import mlrun
+
+@dsl.pipeline()
+def kfpipeline(alias="default", host="milvus"):
+    project = mlrun.get_current_project()
+    
+    # First we create the data
+    create_data_step = mlrun.run_function("create-data", returns=["data"])
+
+    # Then we insert that data into the Vector DB
+    insert_data_into_milvus_step = mlrun.run_function("insert-data-into-milvus", inputs={"data_uri":create_data_step.outputs['data'], "alias":"default","host":"milvus", "port":"19530"}).after(create_data_step)
+    
+    # Then we deploy a nuclio function that we can invoke to get similar books
+    project.deploy_function("inference-function", env={"alias":alias, "host":host, "port":"19530"}).after(insert_data_into_milvus_step)
+```
+
+Set the workflow and run it.
+```python
+project.set_workflow(name='populate-milvus-workflow', workflow_path="workflow.py")
+project.save()
+project.run(name="populate-milvus-workflow", arguments={"alias":"default", "host":"milvus"}, watch=True)
+```
+
+![](_static/populate-milvus-workflow-steps.png)
+
+In the first step "create-data", the following artifact was created
+```python
+project.get_artifact("create-data_data").to_dataitem().as_df()
+```
+![](_static/create-data-artifact.png)
+
+We can now get the created nuclio function and invoke it, infering the vector DB and getting books similar to the query
```

Comment:
Now you can get the created Nuclio function and invoke it, inferring the vector DB and getting books similar to the query